test/test_device_batch_memcpy.cu

```diff
@@ -499,6 +500,115 @@ void TestVectorizedCopy()
   }
 }
 
+template <uint32_t NUM_ITEMS, uint32_t MAX_ITEM_VALUE, bool PREFER_POW2_BITS>
+__global__ void TestBitPackedCounterKernel(uint32_t *bins,
+                                           uint32_t *increments,
+                                           uint32_t *counts_out,
+                                           uint32_t num_items)
+{
+  using BitPackedCounterT =
+    cub::detail::BitPackedCounter<NUM_ITEMS, MAX_ITEM_VALUE, PREFER_POW2_BITS>;
+  BitPackedCounterT counter{};
+  if (threadIdx.x == 0 && blockIdx.x == 0)
```

Comment:
The computation is guarded, but writing output results is not. Since you are launching the kernel with a single kernel any way, let's remove this condition.